core/src/main/scala/org/bykn/bosatsu/Evaluation.scala

```diff
@@ -514,7 +528,8 @@ case class Evaluation(pm: PackageMap.Inferred, externals: Externals) {
             // this isn't great, but since we fully compile, even when
             // we don't use a branch, we hit this now
             (Scoped.unreachable, Type.IntType)
-          case Some(NameKind.Let(expr)) =>
+          case Some(NameKind.Let(recursive, expr)) =>
+            // TODO handle recursion
```

Comment:
write a test that this fails, and fix. I think it should use the same Scope.recursive as above.